projects/maker/index.js

```diff
@@ -49,57 +48,45 @@ async function getJoins(block) {
 }
 
 async function tvl(timestamp, block) {
-  let balances = {};
-  balances[MakerSCDConstants.WETH_ADDRESS] = (await sdk.api.erc20.balanceOf({
-    block,
-    target: MakerSCDConstants.WETH_ADDRESS,
-    owner: MakerSCDConstants.TUB_ADDRESS
-  })).output;
+  const toa = [
+    [MakerSCDConstants.WETH_ADDRESS, MakerSCDConstants.TUB_ADDRESS,],
+  ]
 
-  if (block >= MakerMCDConstants.STARTBLOCK) {
+  const blacklistedJoins = [
+    '0x7b3799b30f268ba55f926d7f714a3001af89d359',
+    '0x41ca7a7aa2be78cf7cb80c0f4a9bdfbc96e81815',
+  ]
+  if (block > MakerMCDConstants.STARTBLOCK) {
     let joins = await getJoins(block);
+    joins = joins.filter(i => !blacklistedJoins.includes(i))
 
-    await Promise.all(joins.map(async join => {
-      try {
-        const gem = (await sdk.api.abi.call({
-          block,
-          target: join,
-          abi: MakerMCDConstants.gem
-        })).output;
-        const balance = (await sdk.api.erc20.balanceOf({
-          target: gem,
-          owner: join,
-          block
-        })).output;
-
-        const symbol = join === "0xad37fd42185ba63009177058208dd1be4b136e6b"?"SAI": await sdk.api.erc20.symbol(gem)
-        if (symbol.output === "UNI-V2") {
-          await unwrapUniswapLPs(balances, [{
-            token: gem,
-            balance
-          }],
-            block)
-        } else {
-          sdk.util.sumSingleBalance(balances, gem, balance);
-        }
-      } catch (e) {
-        try{
-          if(join !== "0x7b3799b30f268ba55f926d7f714a3001af89d359"){
-            await sdk.api.abi.call({
-              block,
-              target: join,
-              abi: MakerMCDConstants.dog
-            })
-          }
-          return
-        } catch(e){
-          throw new Error("failed gem() and dog() on "+join)
-        }
+    const { output: gems } = await sdk.api.abi.multiCall({
+      abi: MakerMCDConstants.gem,
+      block, calls: joins.map(i => ({ target: i })),
+    })
+    const dogCalls = []
+    gems.forEach(({ success, output, input: { target } }) => {
+      if (!success) {
```

Comment:
better to avoid this because of rpc failures, need to code it so an incorrect failure doesnt rug some collateral

Reply:
but if something should succeed here but fails, then it will fail dog call as well and the adapter throws an error